server/src/graphql/mutations/createAppGithub.ts

```diff
@@ -32,43 +34,42 @@ export const createAppGithub: MutationResolvers['createAppGithub'] = async (
     },
   });
 
+  const user = await prisma.user.findUnique({
+    where: {
+      id: userId,
+    },
+  });
+
   const isAppNameTaken = !!apps[0];
 
   if (isAppNameTaken) {
     throw new Error('App name already taken');
   }
 
-  const octokit = new Octokit({});
-
-  const repoData = getRepoData(input.gitRepoUrl);
-
-  const repo = await octokit.repos.get({
-    owner: repoData.owner,
-    repo: repoData.repoName,
+  const auth = createAppAuth({
+    appId: config.githubAppId,
+    privateKey: config.githubAppPem,
+    clientId: config.githubAppClientSecret,
+    clientSecret: config.githubAppClientSecret,
   });
 
-  if (!repo) {
-    throw new Error(
-      `No repository found for ${repoData.owner} with name ${repoData.repoName}`
-    );
-  }
-
-  const branch = await octokit.repos.getBranch({
-    owner: repoData.owner,
-    repo: repoData.repoName,
-    branch: input.branchName ? input.branchName : 'main',
-  });
+  const installationAuthentication = (await auth({
+    type: 'installation',
+    installationId: input.githubInstallationId,
+  })) as AppAuthentication;
 
-  if (!branch.url) {
-    throw new Error(
-      `There's no ${input.branchName} branch or main branch for this repository`
-    );
-  }
+  const repoData = getRepoData(input.gitRepoUrl);
 
   const ssh = await sshConnect();
 
   const dokkuApp = await dokku.apps.create(ssh, input.name);
 
+  const dokkuAuth = await dokku.git.auth({
```

Comment:
do we have to do this here? My understanding is that a token is valid for 24h only, so we should rather call this every time we receive a webhook push event.

Reply:
Good point, will add this to webhook handler too. I think we should leave it here as well as we need to be authenticated before the first deploy as well.

Reply:
Yeah actually we could probably move it in the queue, that way we would have this logic in only one place.

Reply:
Yeah. moving this to queue makes a lot of sense. 

On it.